src/debuggerCommunicationServer.ts

```diff
@@ -57,14 +82,23 @@ export class DebuggerCommunicationServer {
 
     private initEventsHandlers(): void {
         this.serverIo.on("connection", (socket: any) => {
-            console.log("Connection received");
-
-            socket.on("updateState", (data: any) => {
+            socket.on(DEBUGGER_MESSAGES.LISTENER.UPDATE_STATE, (data: any) => {
                 this.handleState(data);
+                this.serverIo.emit(
+                    DEBUGGER_MESSAGES.EMITTER.RECEIVED_STATE,
+                    {}
+                );
+            });
+            socket.on(DEBUGGER_MESSAGES.LISTENER.RECEIVED_STATE, () => {
+                this.isWaitingResponse = false;
```

Comment:
We can alternatively only assign this to false if `this.currentCall.length == 0` -- then we wouldn't need to update this value inside `currentCall()` again.

Reply:
good suggestion! Removed it from current call and setting it to true if this.currentcall.lenth>0. If it's 0, then i'm putting it to false since there's no more calls

Reply:
I _think_ we don't need to assign it to true on the listener? Since we won't assign it to false elsewhere. Anyway, it's up to you if you think that's a good sanity check!